app/javascript/components/mentoring/mentor_inbox.jsx

```diff
@@ -17,6 +18,11 @@ function reducer(state, action) {
         ...state,
         query: { ...state.query, sort: action.payload.sort },
       }
+    case 'filter.changed':
+      return {
+        ...state,
+        query: { ...state.query, filter: action.payload.filter, page: 1 },
+      }
     default:
       if (process.env.NODE_ENV === 'development') {
         throw 'Unknown action type'
```

Comment:
Always throw errors, and add debuggable information to the error (preferably actionable).

```js
new Error(`Unknown action type: ${action.type}`)
```

Reply:
I've fixed this in #78